src/runtime_src/core/pcie/driver/linux/xocl/subdev/fmgr.c

```diff
@@ -157,13 +158,22 @@ static int fmgr_probe(struct platform_device *pdev)
 	struct xfpga_klass *obj = kzalloc(sizeof(struct xfpga_klass), GFP_KERNEL);
 	if (!obj)
 		return -ENOMEM;
-
+    /* TODO: Remove old fpga_mgr_register call as soon as Linux < 4.18 is no
+     * longer supported.
+     */
+#if defined(FPGA_MGR_SUPPORT) && (LINUX_VERSION_CODE >= KERNEL_VERSION(4,18,0))
+	struct fpga_manager *mgr = platform_get_drvdata(pdev);
```

Comment:
where is the drv data you're getting here set?

Reply:
Shouldn't `platform_get_drvdata` set the drv data? That's how I understood the equivalent call in `fmgr_remove` (line 186).

Reply:
oh? why do we ever need to call platform_set_drvdata(), if platform_get_drvdata() actually set it?

Reply:
You're right, sorry. I misinterpreted the call in `fmgr_remove`.

Reply:
This should be fixed now, please review again :)